src/app/api/feedback/route.js

```diff
@@ -0,0 +1,59 @@
+import { NextResponse } from 'next/server';
+import { z } from 'zod';
+import { prisma } from '../../../lib/db';
+
+const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+const phoneRegex = /^\+380\d{9}$/;
+
+const Feedback = z
+  .object({
+    name: z
+      .string({
+        required_error: 'Name is required',
+        invalid_type_error: 'Name must be a string',
+      })
+      .max(128, {
+        message: 'Name must not be longer than 128 characters',
+      }),
+    phone: z
+      .string({
+        required_error: 'Phone number is required',
+        invalid_type_error: 'Phone number must be a string',
+      })
+      .refine(val => phoneRegex.test(val), {
+        message: 'Please, enter phone number in format +380XXXXXXXXX',
+      }),
+    callAgain: z.boolean({
+      required_error: 'callAgain is required',
+      invalid_type_error: 'callAgain must be a boolean',
+    }),
+    email: z
+      .string()
+      .refine(val => emailRegex.test(val) && val.length <= 254, {
+        message: 'Please, enter valid email',
+      }),
+    message: z
+      .string({
+        required_error: 'Message is required',
+        invalid_type_error: 'Message must be a string',
+      })
+      .max(320, { message: 'Message must not be longer than 320 characters' }),
+  })
+  .partial({ email: true });
+
+export async function POST(request) {
+  try {
+    const feedbackData = await request.json();
+    const validatedFeedback = Feedback.safeParse(feedbackData);
+    if (!validatedFeedback.success) {
+      return NextResponse.json(
+        { errors: validatedFeedback.error.flatten().fieldErrors },
+        { status: 400 },
+      );
+    }
+    const feedback = await prisma.feedback.create({ data: feedbackData });
+    return NextResponse.json(feedback);
+  } catch (err) {
+    return NextResponse.json({ error: err }, { status: 500 });
```

Comment:
🟡 i think this `try-catch` could be limited in scope to be only around prisma call

Reply:
> 🟡 i think this `try-catch` could be limited in scope to be only around prisma call

line 46 can throw errors too

Reply:
true, can it be handled by zod too?

Reply:
As I understand, zod is for schema validation only and not for http requests invalidation, so I dont think it can be handled with zod. But I may be wrong, I think we need to ask Taras about it

Reply:
yeah, i mean if zod already validates request body, maybe it can validate that body is proper json too

Reply:
As a suggestion, you can catch both zod errors and server errors in the catch block. 

```
export async function POST(request) {
  try {
    const feedbackData = await request.json();

    Feedback.parse(feedbackData); // have to use parse() as it throws an error 

    await prisma.feedback.create({ data: feedbackData });

    return NextResponse.json(feedbackData);
  } catch (err) {
    if (err instanceof z.ZodError) {
      return NextResponse.json(
        {
          error: err.flatten().fieldErrors,  
        },
        { status: 400 },
      );
    }

    return NextResponse.json({ error: err.message }, { status: 500 });
  }
}
```

Reply:
@alex-popov-tech @dmitrenko-v maybe we should think about some general error handler that will be used across the whole app? 

```
const handleError = (errorMessage, status) => {
  return NextResponse.json({ error: errorMessage }, { status });
};
```